src/main/java/com/box/sdk/BoxStoragePolicy.java

```diff
@@ -0,0 +1,175 @@
+
+    /**
+     * Gets information for a Box Storage Policy with optional fields.
+     *
+     * @param fields the fields to retrieve.
+     * @return info about this item containing only the specified fields, including storage policy.
+     */
+    public BoxStoragePolicy.Info getInfo(String... fields) {
+        QueryStringBuilder builder = new QueryStringBuilder();
+        if (fields.length > 0) {
+            builder.appendParam("fields", fields);
+        }
+        URL url = STORAGE_POLICY_WITH_ID_URL_TEMPLATE.buildWithQuery(this.getAPI().getBaseURL(), builder.toString(),
+                this.getID());
+
+        BoxAPIRequest request = new BoxAPIRequest(this.getAPI(), url, "GET");
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        return new Info(response.getJSON());
+    }
+
+    /**
+     * Returns all BoxStoragePolicy with specified fields.
+     *
+     * @param api    the API connection to be used by the resource.
+     * @param fields the fields to retrieve.
+     * @return an iterable with all the storage policies met search conditions.
+     */
+    public static Iterable<BoxStoragePolicy.Info> getAll(final BoxAPIConnection api, String... fields) {
+
+        return getAll(api, DEFAULT_LIMIT, fields);
+    }
+
+    /**
+     * Returns all BoxStoragePolicy with specified fields.
+     *
+     * @param api    the API connection to be used by the resource.
+     * @param limit  the limit of items per single response. The default is 100.
+     * @param fields the fields to retrieve.
+     * @return an iterable with all the storage policies met search conditions.
+     */
+    public static Iterable<BoxStoragePolicy.Info> getAll(final BoxAPIConnection api, int limit, String... fields) {
+
+        QueryStringBuilder builder = new QueryStringBuilder();
+        if (fields.length > 0) {
+            builder.appendParam("fields", fields);
+        }
+
+        URL url = STORAGE_POLICY_URL_TEMPLATE.buildWithQuery(api.getBaseURL(), builder.toString());
+        return new BoxResourceIterable<BoxStoragePolicy.Info>(api, url, limit) {
+
+            @Override
+            protected BoxStoragePolicy.Info factory(JsonObject jsonObject) {
+                BoxStoragePolicy storagePolicy = new BoxStoragePolicy(api, jsonObject.get("id").asString());
+
+                return storagePolicy.new Info(jsonObject);
+            }
+        };
+    }
+
+    /**
+     * Checks if there is already a Storage Policy Assignment and creates one if one does not exist.
+     * @param api                the API connection to be used by the resource.
+     * @param storagePolicyID    the ID of the Storage Policy you want to assign to user.
+     * @param userID             the ID of the user you want to assign the Storage Policy to.
+     * @return information about this {@link BoxStoragePolicyAssignment}.
+     */
+    public static BoxStoragePolicyAssignment.Info assign(BoxAPIConnection api, String storagePolicyID, String userID) {
```

Comment:
There's already a static one of these on `BoxStoragePolicyAssignment` — this one should probably be an instance method